src/promptflow-core/promptflow/_utils/multimedia_utils.py

```diff
@@ -2,278 +2,497 @@
+
+    def _resolve_image_path(self, input_dir: Path, image_dict: dict):
+        """Resolve image path to absolute path in image dict"""
+
+        input_dir = input_dir.parent if input_dir.is_file() else input_dir
+        if self.is_multimedia_dict(image_dict):
+            for key in image_dict:
+                _, resource = ImageProcessor.get_multimedia_info(key)
+                if resource == "path":
+                    image_dict[key] = str(input_dir / image_dict[key])
+        return image_dict
+
+    def resolve_multimedia_data_recursively(self, input_dir: Path, value: Any):
+        process_funcs = {self.is_multimedia_dict: partial(self._resolve_image_path, input_dir)}
+        return self._process_multimedia_dict_recursively(value, process_funcs)
+
+    def persist_multimedia_data(
+        self, value: Any, base_dir: Path, sub_dir: Path = None, use_absolute_path=False, inplace: bool = False
+    ):
+        pfbytes_file_reference_encoder = (
+            self.get_file_reference_encoder(base_dir, sub_dir, use_absolute_path=use_absolute_path)
+            if base_dir
+            else None
+        )
+        serialization_funcs = {Image: partial(Image.serialize, **{"encoder": pfbytes_file_reference_encoder})}
+        return _process_recursively(value, process_funcs=serialization_funcs, inplace=inplace)
+
+    @staticmethod
+    def convert_multimedia_data_to_base64_dict(value: Any):
+        def convert_pfbytes_to_base64_dict(obj: PFBytes):
+            return {f"data:{obj._mime_type};base64": obj.to_base64()}
+
+        to_base64_funcs = {PFBytes: convert_pfbytes_to_base64_dict}
+        return _process_recursively(value, process_funcs=to_base64_funcs)
+
+
+class OpenaiVisionMultimediaProcessor(MultimediaProcessor):
+    @staticmethod
+    def is_multimedia_dict(multimedia_dict: dict):
+        if len(multimedia_dict) != 2:
+            return False
+        if "type" not in multimedia_dict:
+            return False
+        image_type = multimedia_dict["type"]
+        if image_type in ["image_url", "image_file"] and image_type in multimedia_dict:
+            return True
+        return False
 
-def create_image(value: any):
-    if isinstance(value, PFBytes):
-        return value
-    elif isinstance(value, dict):
-        if is_multimedia_dict(value):
-            return _create_image_from_dict(value)
+    @staticmethod
+    def _create_image_from_dict(image_dict: dict):
+        image_type = image_dict["type"]
+        if image_type == "image_url":
+            if ImageProcessor.is_base64(image_dict["image_url"]["url"]):
+                return ImageProcessor.create_image_from_base64(image_dict["image_url"]["url"])
+            elif ImageProcessor.is_url(image_dict["image_url"]["url"]):
+                return ImageProcessor.create_image_from_url(image_dict["image_url"]["url"])
+            else:
+                raise InvalidImageInput(
+                    message_format=f"Invalid image url: {image_dict['image_url']}."
```

Comment:
Should be `image_dict['image_url']['url']`?

This part is repeated multiple times. Should we assign it to a variable first for better readability?

Reply:
Done.